File: WordCount/Python/wordCount.py


from pyspark import SparkConf
from pyspark import SparkContext

sc = SparkContext("local", "wordCount")

text_file = sc.textFile(inputFile) //Load our input data.
counts = text_file.flatMap(lambda line: line.split(" ")) \
             .map(lambda word: (word, 1)) \
             .reduceByKey(lambda a, b: a + b)
counts.saveAsTextFile(outputFile)

File: WordCount/Java/java8/wordCount.java


import java.util.Arrays;
import org.apache.spark.SparkConf;
import org.apache.spark.api.java.JavaRDD;
import org.apache.spark.api.java.JavaPairRDD;
import org.apache.spark.api.java.JavaSparkContext;
import scala.Tuple2;

public class JavaSpark {

    public static void main(String args[]) throws Exception {
        String inputFile = args[0];
        String outputFile = args[0];
        SparkConf conf = new SparkConf().setAppName("WordCount");
        JavaSparkContext sc = new JavaSparkContext(conf);

        JavaRDD<String> input = sc.textFile(inputFile);
        JavaRDD<String> words = input.flatMap(s -> Arrays.asList(s.split(" ")));
        JavaPairRDD<String, Integer> counts = words.mapToPair(x -> new Tuple2<String, Integer>(x, 1)).reduceByKey((x, y) -> x + y);
        counts.saveAsTextFile(outputFile);

    }
}


